membrane_rtc_engine/lib/membrane_rtc_engine/engine.ex

```diff
@@ -748,7 +764,6 @@ defmodule Membrane.RTC.Engine do
 
     tracks_msgs = build_track_removed_actions(tracks, endpoint_id, state)
     track_ids = Enum.map(tracks, & &1.id)
-    track_tees = tracks |> Enum.map(&get_track_tee(&1.id, ctx)) |> Enum.reject(&is_nil(&1))
```

Comment:
This was correct behavior. In general the flow should look like this:
1. endpoint publishes `removed_tracks`
2. engine removes tee
3. endpoint get handle_pad_removed(:output) and clean ups its children
4. other endpoints get removed_tracks and then handle_pad_removed(:input) and clean ups their children